### PR TITLE
finished most of it

### PR DESCRIPTION
**Pull request by ChCosmin**

### CODE REVIEW

File: THIS/evaluations.css [hunk 1]
```diff
@@ -1,13 +1,124 @@
...
+	text-align: right;
+	padding-right: 5px;
+}
+
+button{
+	width:40px;
+	height:40px;
+	margin-left: 5px;
+	background-color:#fcfcfc;
+	background-image: url('plus.png');
```
Comment (softvision-alexcarcea) [inline THIS/evaluations.css]: Nicely done! Good separation of data / model vs presentation / style. Any way to contract these background-related properties, though?

File: THIS/evaluations.css [hunk 2]
```diff
@@ -1,13 +1,124 @@
...
+/* --------FOOTER----------- */
+footer{ 
+	height: 80px;
+	background-color: #e6e6e6;
+	text-align: center;
+	display:flex;
+	justify-content:flex-end;
+	align-items:center;
+	position: absolute;
+    bottom: 0px;
```
Comment (softvision-alexcarcea) [inline THIS/evaluations.css]: What if I told you that this should be `0%` instead? What about `0em`? Would it matter?

File: THIS/evaluations.css [hunk 3]
```diff
@@ -1,13 +1,124 @@
...
+    width: 100%;
+}
+
+footer > span {
```
Comment (softvision-alexcarcea) [inline THIS/evaluations.css]: In such cases, it's often a better idea (for maintenance in the long run) to assign specific classes on the targeted elements and style off of those. The block-element-modifier methodology nicely complements this pattern, by allowing you to be sufficiently generic, while also being specific enough to only target the elements you care about.

File: THIS/evaluations.html [hunk 4]
```diff
@@ -7,14 +7,15 @@
...
-			<img id="logo" src="svlogo.png" alt="softvision logo">
-		<nav>
-			<a href="evaluations.html">Evaluations</a>
-			<a href="newEvaluation.html">New Evaluation</a>
-			</ul>
-		</nav>
-		<a href="Logout.html">Logout</a>
+	<header>		
+	<nav>
+		<a href="#home"><img id="logo" src="svlogo.png" alt="softvision logo"></a>
```
Comment (softvision-alexcarcea) [inline THIS/evaluations.html]: Ah, a proper use of the `alt` attribute! Kudos!

File: THIS/new_evaluations.html [hunk 5]
```diff
@@ -484,7 +484,9 @@
 					</ul>
 				</fieldset>	
 			</section>
-			<button type="submit" id="Submit" value="Submit">Submit</button>		
+			<section id="button">
+				<button type="submit" id="Submit" value="Submit">SUBMIT</button>
```
Comment (softvision-alexcarcea) [inline THIS/new_evaluations.html]: What if I told you that you could affect the casing of a piece of text through CSS? Thus, you could keep the same data intact, but just display it differently, based on the applicable styling rulesets?

File: THIS/login.css [hunk 6]
```diff
@@ -9,7 +9,7 @@ header {
 	margin-top:60px;
 }
 
-header > form {
+header form {
```
Comment (softvision-alexcarcea) [inline THIS/login.css]: Cleanup step 2: consider rewriting this using BEM.

File: THIS/new_evaluations.css [hunk 7]
```diff
@@ -71,6 +74,8 @@ nav *:active{
 	color:#d9d9d9;
 	font-style: normal;
 }
+
+/* ---------TECHNICAL LEVEL--------- */
 .sec1 ul li {
```
Comment (softvision-alexcarcea) [inline THIS/new_evaluations.css]: Well, this is an interesting selector. Could you explain its purpose, please?

File: THIS/evaluations.css [hunk 8]
```diff
@@ -1,13 +1,124 @@
-header {
-	height: 200px;
-	background-color: red;
+html {
+	font-family: "Arial", Helvetica, sans-serif;
```
Comment (ChCosmin) [inline THIS/evaluations.css]: Backup fonts in case the first one, second one, and so on are not found (even tho Arial is a pretty common font so chances of it not being found are low i guess, but it's a good practice non the less)

File: THIS/evaluations.css [hunk 9]
```diff
@@ -1,13 +1,124 @@
...
+	background-color: #fcfcfc;
 }
 
-img {
-	height: 18%;
+/* ----------HEADER-------------*/
+header{
```
Comment (ChCosmin) [inline THIS/evaluations.css]: fair enough, not sure why i forgot to add the same style to everything, must've been due to a bit of frustration and constant retrying of solutions that i missed out on the consistency, thanks for pointing that out

File: THIS/evaluations.css [hunk 10]
```diff
@@ -1,13 +1,124 @@
...
+	text-decoration: none;
 }
 
-header *{
-	display: inline-block;
-	float: left;
+nav {
+	font-size: 25px;
+	height:100%;
+	display:flex;
```
Comment (ChCosmin) [inline THIS/evaluations.css]: font-size: 25px = wanted the font to be of that size @_@
height: 100% = wanted the nav to be as tall as the header (so the blue underline would be in the right place when a link is active)
display: flex = wanted the nav elements to be distributed evenly over the header, giving each item a flex:<number value> attribute to adjust the spacing of each item/group of items

File: THIS/evaluations.css [hunk 11]
```diff
@@ -1,13 +1,124 @@
...
+	height:100%;
+	display:flex;	
+	align-items: center;
+}
+
+nav *{
+	flex:1;
+}
+
+nav *:visited{
```
Comment (ChCosmin) [inline THIS/evaluations.css]: since all items in the header&nav are a tags and have preset properties like being underlines, or change color once visited i didn't want that for any of the a tags in the nav, plus when i added my own css to all of the a tags (like the blue underline when being clicked) i didn't want that for the logo or logout items, if there were more than 4x a tags this seemed helpful

File: THIS/evaluations.css [hunk 12]
```diff
@@ -1,13 +1,124 @@
...
+	color:#000000;
+}
+
+#logo {
+	width:50%;
+	padding: 15px 120px 15px 30px;
+}
+
+#menu {
+	width:1232px;
```
Comment (ChCosmin) [inline THIS/evaluations.css]: i remember having heard my coleague saying Vali set the form elements that contained those many select tags to be of that width which with the help of adding the value of width:33% to each li tag, it would arange them in that specific 3 colum element pattern.

File: THIS/evaluations.css [hunk 13]
```diff
@@ -1,13 +1,124 @@
...
+	flex:3.5;
+	height:50%;
+}
+
+#menu a{
```
Comment (ChCosmin) [inline THIS/evaluations.css]: as far as i remember, you use the ID when you have a single element on the page that will take the style (that being said, i looked over and it's uses style that other elements already use) but when i first put it as an id, i didn't expect to use some of the same styles.

File: THIS/evaluations.css [hunk 14]
```diff
@@ -1,13 +1,124 @@
...
+}
+
+#logout{
+	text-decoration: none; 
+	text-align: right;
+	padding-right:20px;
+}
+
+/* ------------TABLE---------- */
+section{
```
Comment (ChCosmin) [inline THIS/evaluations.css]: as far as their width and centering with margin goes yes. i guess i could've put all of the sections in a wrapper and make the wrapper that centered instead of each element individually, but that didn't cross my mind when i styled it also idk if doing that will break some stuff inside that wrapper. but i think the wrapper thing would've worked somehow aswell, feel free to corect me :)

Comment (ChCosmin) [inline THIS/evaluations.css]: reusability pros and cons? well unless new form fields won't have the same pattern/format as all the rest, then that generalization might make it more difficult to stylize (general speaking). of course you could always make it a different class and override some attributes but that doesn't sound efficient.

if the new form fields respects the same format as the rest, then it helps because you don't need to waste time stylizing something of the same type as the others, you just put the tag that's already been stylized to look like the others thus saving time

File: THIS/evaluations.css [hunk 15]
```diff
@@ -1,13 +1,124 @@
...
+th {
+	background-color:#03A9F4;
+	color:#ffffff;
+}
+
+tr {
+	border-bottom:1px solid black;	
+}
+
+td:nth-child(4){
```
Comment (ChCosmin) [inline THIS/evaluations.css]: since there are 4 rows, each with 4 columns, i only wanted to target the 4th cell(td) of each row to be closer to the edge of the table

File: THIS/evaluations.css (hunk 1 above)
Comment (ChCosmin) [inline THIS/evaluations.css]: ah didn't know you can put it all in one line
background: bg-color bg-image position/bg-size bg-repeat bg-origin bg-clip bg-attachment initial|inherit;, good to know thanks

File: THIS/evaluations.css (hunk 3 above)
Comment (ChCosmin) [inline THIS/evaluations.css]: sry but not sure i understood what you meant. please rephrase

File: THIS/evaluations.css (hunk 2 above)
Comment (ChCosmin) [inline THIS/evaluations.css]: em is used based on the current form as far as i know.
% is relative to the height/width of the parent element so im guessing the % WOULD be the smarter choice

File: THIS/new_evaluations.html (hunk 5 above)
Comment (ChCosmin) [inline THIS/new_evaluations.html]: not sure i understood what you meant by casing of a piece of text through css. care to explain more and/or give examples? this sounds interesting

File: THIS/login.css (hunk 6 above)
Comment (ChCosmin) [inline THIS/login.css]: not...sure...how...

File: THIS/new_evaluations.css (hunk 7 above)
Comment (ChCosmin) [inline THIS/new_evaluations.css]: which one? the sec1 class?  i wrote technical level because that's the name of the form field so i find the code for it faster

File: THIS/evaluations.css (hunk 8 above)
Comment (softvision-alexcarcea) [inline THIS/evaluations.css]: Yeppers, a best practice, indeed. Nicely done!

File: THIS/evaluations.css (hunk 11 above)
Comment (softvision-alexcarcea) [inline THIS/evaluations.css]: Fair, but in this particular case, since `:visited` / `:active` only apply to anchor tags, I'd say it's best to be a bit more specific, if for no other reason than simply because these pseudo-classes would not make sense for other elements.

File: THIS/evaluations.css (hunk 12 above)
Comment (softvision-alexcarcea) [inline THIS/evaluations.css]: Ah, so this is a sort of home-brewed 12 column grid?

Comment (ChCosmin) [inline THIS/evaluations.css]: that's a good wait to put it, vali just showed us how to make/use/think a grid system so that should help me when i try to remake the project

File: THIS/evaluations.css (hunk 13 above)
Comment (softvision-alexcarcea) [inline THIS/evaluations.css]: For a bit of context, the reason why I'm asking is that we want to force you to think about what types of selectors you're styling against. Going for the ID is probably the easiest option, and the one I've used myself most often in my early days as a front-end dev. The downside of doing this is that it becomes difficult to override styles that are keyed off of an ID, which makes this an ill-suited candidate for larger applications. Bottom line: only use IDs when you absolutely must and you are certain that no other element would need to share that style and that you will never need to override that style.

File: THIS/evaluations.css (hunk 14 above)
Comment (softvision-alexcarcea) [inline THIS/evaluations.css]: Sounds like you have a pretty good grasp on what's what. Carry on. ;)

File: THIS/evaluations.css (hunk 2 above)
Comment (softvision-alexcarcea) [inline THIS/evaluations.css]: Then, let me ask you this: is `0%`, specifically, different than `0px`? If so, how? If not, would we be able to get away with just a plain ol' `bottom: 0;`? :)

File: THIS/evaluations.css (hunk 3 above)
Comment (softvision-alexcarcea) [inline THIS/evaluations.css]: When you find yourself more tightly coupling the styling to the underlying markup, one of the first things I'd like you to think about is if there are any other alternatives that would decrease cohesion, such that if you changed the markup, you wouldn't necessarily need to update the styling. While this might suffice in this particular case, ask yourself if a class would have been a suitable alternative. Would it do the job better? Would it be the same? Would it be worse? That depends on the context - there is no magic formula, aside from thinking it through. :)

File: THIS/new_evaluations.html (hunk 5 above)
Comment (softvision-alexcarcea) [inline THIS/new_evaluations.html]: [text-transform](https://developer.mozilla.org/en/docs/Web/CSS/text-transform) Shh, don't tell anyone! :P But no, really, feel free to share! ;)

File: THIS/new_evaluations.css (hunk 7 above)
Comment (softvision-alexcarcea) [inline THIS/new_evaluations.css]: My pain point is around that `1` at the end. To me, it sort of implies that there'll be more of these. I'm sure you'd be able to find a better suited name that's both generic enough and suggestive enough with respect to its purpose and usage.

File: THIS/evaluations.css (hunk 2 above)
Comment (ChCosmin) [inline THIS/evaluations.css]: not entirely sure but i think when you put 0px it's stricly 0px, when you put 0% i THINK it can also (maybe) encompass values between 0-0.9999999etc %. It's either THAT or no there is no difference if the value is specifically zero.

File: THIS/evaluations.css (hunk 3 above)
Comment (ChCosmin) [inline THIS/evaluations.css]: your english and terminology is a bit too much for my knowledge, but lemme see if i got it, basically what you want to say is that i should generally go for styling classes attached to tags instead of the tags themselves? or did i missed the point again? explain a bit more in layman terms please

File: THIS/new_evaluations.html (hunk 5 above)
Comment (ChCosmin) [inline THIS/new_evaluations.html]: aah now i get what you were referring to, got it, yeah, that didn't cross my mind, good reminder

File: THIS/new_evaluations.css (hunk 7 above)
Comment (ChCosmin) [inline THIS/new_evaluations.css]: for some reason i ended up not using sec2/sec3. Basically the form on new_eval had more or less 3 types of fields (radio button/sec1, textareas/sec2, dropdownmenu/sec3) but instead of styling the classes i forgot and styled the tags which i shouldn't have (will remake the site)

File: THIS/evaluations.css (hunk 2 above)
Comment (softvision-alexcarcea) [inline THIS/evaluations.css]: So, if we went out for a beer and I said that you can have `0%` of my beer, are you saying you'd actually have reasonable expectations that I'd give you anywhere up to, but not including, `1%`? Dunno 'bout you, but I'm pretty fond of my beer and don't usually share... :P

**tl; dr**: `0%` / `0px` / `0em` / `0whatever` is precisely `0`, in any circumstance. ;)

File: THIS/evaluations.css (hunk 3 above)
Comment (softvision-alexcarcea) [inline THIS/evaluations.css]: Styling against classes is usually the more flexible option, yes, since it provides that good balance between abstraction and concreteness. There may be times when you'll want to style against tags as well, but especially when working on large-scale applications, that might have unknown and unwanted implications, so using classes for styling is usually the safer option.
